packages/slate/src/interfaces/path-ref.ts

```diff
@@ -21,18 +22,24 @@ export interface PathRefInterface {
 
 // eslint-disable-next-line no-redeclare
 export const PathRef: PathRefInterface = {
-  transform(ref: PathRef, op: Operation): void {
+  transform(ref: PathRef, op: Operation): (() => void) | void {
     const { current, affinity } = ref
 
     if (current == null) {
       return
     }
 
+    const prevPath = ref.current ? [...ref.current] : ref.current
     const path = Path.transform(current, op, { affinity })
+
     ref.current = path
 
     if (path == null) {
       ref.unref()
     }
+
+    if (path && prevPath ? !Path.equals(path, prevPath) : path !== prevPath) {
+      return () => ref.onChange(path)
+    }
```

Comment:
I found this ternary quite hard to read. I wasn't sure if it was `(path && prevPath) ? ...` or `path && (prevPath ? ...)` until I checked in an AST explorer.

If you go with my suggestion above of return a boolean instead, how about splitting this into two if statements to be more readable?
```suggestion
    if (path && prevPath) {
      return !Path.equals(path, prevPath)
    }
    
    return path !== prevPath
```